src/info/langs/mod.rs

```diff
@@ -66,146 +73,13 @@ fn get_statistics(
     globs_to_exclude: &[String],
     language_types: &[LanguageType],
     include_hidden: bool,
-) -> tokei::Languages {
-    let mut languages = tokei::Languages::new();
-    let filtered_languages = filter_languages_on_type(language_types);
-
-    let tokei_config = tokei::Config {
-        types: Some(filtered_languages),
-        hidden: Some(include_hidden),
-        ..tokei::Config::default()
-    };
-    let ignored: Vec<&str> = globs_to_exclude.iter().map(AsRef::as_ref).collect();
-    languages.get_statistics(&[&dir], &ignored, &tokei_config);
-    languages
+) -> Result<gengo::Analysis, Box<dyn Error>> {
+    // TODO Determine best way to ignore files (and if that should continue to be handled by onefetch)
```

Comment:
It would be nice to have the language type filter and glob filtering for a V1

Reply:
or is this going to be handled via `.gitattributes`?

Reply:
That depends on the "file source." Right now the `Git` file source is highly opinionated and would use `.gitattributes` for ignoring files, overrides, etc.

******

I plan `Directory` file source being much less opinionated (can't get much more generic than just reading a folder :slightly_smiling_face:). The functionality isn't there yet, but I was considering adding ways to configure excluding/including files. Right now `.gitignore` and `.ignore` files would ignore files via the `ignore` crate's default behavior. I was hoping to discuss the usage sometime, but it would probably either be like this:
```rust
let file_source = Directory::with_config("./", Config { ignored_files: Some(vec![]) });
```
or like this:
```rust
let file_source = Directory::new("./").extend_ignored_files(&[]);
```

Reply:
By default Gengo marks Programming, Markup, and Query as detectable, and Prose and Data as not detectable. `detectable` is basically a boolean that answers "should this factor in to the stats?" Also, documentation, generated files, and vendored files are not detectable by default. Assuming we use the `Git` file source with `.gitattributes`, this is example usage:
```gitattributes
# Markdown is Prose, but we still want it in the stats
*.md gengo-detectable

# JavaScript is Programming, but we want to exclude it
*.js -gengo-detectable

# The contents of dist/ are not generated, so they will be detectable
dist/* -gengo-generated
# OR
# Even though dist/* is generated, it should still be included in the stats
dist/* gengo-detectable
```

******

Again, the `Git` file source is highly opinionated and tries to behave a lot like github-linguist. The usage with the `Directory` file source is still undecided, and right now it doesn't implement any overrides. Or we could write our own file source (https://github.com/o2sh/onefetch/discussions/1303#discussioncomment-9058916) if we want very specific behavior that wouldn't be suitable for the `gengo` crate.

But, however we implement it, IMO the best way would be to inform gengo what files are and aren't detectable, and then make use of either the `Analysis` (detailed results) or `Summary` (simplified results).

******

tl;dr Yes, gengo would be the one handling included/excluded types, and we'd just pass configuration to it somehow.